src/DotNetWorker.Grpc/IWorkerClientFactory.cs

```diff
@@ -0,0 +1,13 @@
+﻿// Copyright (c) .NET Foundation. All rights reserved.
+// Licensed under the MIT License. See License.txt in the project root for license information.
+
+using System.Threading;
+using System.Threading.Tasks;
+
+namespace Microsoft.Azure.Functions.Worker.Grpc
+{
+    internal interface IWorkerClientFactory
+    {
+        Task<IWorkerClient> StartClientAsync(IMessageProcessor messageProcessor, CancellationToken token);
```

Comment:
What do you think of `ValueTask<IWorkerClient>` here? Considering one of our implementations is synchronous.

But also not a big deal at all as this is called once per worker I think?